--- a/qtbase/src/platformsupport/input/evdevkeyboard/qevdevkeyboardhandler.cpp
+++ b/qtbase/src/platformsupport/input/evdevkeyboard/qevdevkeyboardhandler.cpp
@@ -60,0 +61,8 @@
+#ifndef input_event_sec
+#define input_event_sec time.tv_sec
+#endif
+
+#ifndef input_event_usec
+#define input_event_usec time.tv_usec
+#endif
+
@@ -150,0 +159,2 @@
+    struct timeval tv;
+    ::gettimeofday(&tv, 0);
@@ -152 +162,2 @@
-    ::gettimeofday(&led_ie.time, 0);
+    led_ie.input_event_sec = tv.tv_sec;
+    led_ie.input_event_usec = tv.tv_usec;
